src/bank.rs

```diff
@@ -548,6 +565,12 @@ impl Bank {
                 post_account,
             )?;
         }
+        // Check account subscriptions
+        for ((pre, pubkey), post) in pre_userdata.iter().zip(program_accounts.iter()) {
+            if pre != &post.userdata {
```

Comment:
I think it should be faster to first check if the account pubkey has been subscribed to, then memcmp the account for change second.  This way mem compares each account at the end of each transaction right?